src/main/java/seedu/address/logic/parser/RemoveCommandParser.java

```diff
@@ -29,9 +31,21 @@ public RemoveCommand parse(String args) throws ParseException {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemoveCommand.MESSAGE_USAGE), ive);
         }
 
-        Quantity quantity = new Quantity(argMultimap.getValue(PREFIX_QUANTITY).orElse(""));
+        if (!areQuantityPresent(argMultimap, PREFIX_QUANTITY)) {
+            throw new ParseException(RemoveCommand.MESSAGE_NO_QUANTITY);
+        }
+
+        Quantity quantity = ParserUtil.parseQuantity(argMultimap.getValue(PREFIX_QUANTITY).get());
 
         return new RemoveCommand(index, quantity);
     }
 
+    /**
+     * Returns true if Prefix Quantity does not contains empty {@code Optional} value in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean areQuantityPresent(ArgumentMultimap argumentMultimap, Prefix quantity) {
```

Comment:
would isQuantityPresent be a better name?

Reply:
@halcon-blanco thanks, amended!